ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/KsqlEntity.java

```diff
@@ -43,12 +45,22 @@
 })
 public abstract class KsqlEntity {
   private final String statementText;
+  private final List<KsqlWarning> warnings;
 
   public KsqlEntity(final String statementText) {
+    this(statementText, Collections.emptyList());
+  }
+
+  public KsqlEntity(final String statementText, final List<KsqlWarning> warnings) {
     this.statementText = statementText;
+    this.warnings = warnings == null ? Collections.emptyList() : warnings;
```

Comment:
Take a defensive immutable copy:

```suggestion
    this.warnings = warnings == null ? Collections.emptyList() : ImmutableList.copyOf(warnings);
```